src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -102,7 +100,7 @@ class CargoProjectDescription private constructor(
 
     fun isCrateRoot(file: VirtualFile): Boolean = findTargetForCrateRootFile(file) != null
 
-    fun withAdditionalPackages(additionalPackages: Collection<Pair<String, VirtualFile>>): CargoProjectDescription {
+    fun withAdditionalPackages(additionalPackages: Collection<Pair<String, VirtualFile>>, origin: PackageOrigin): CargoProjectDescription {
```

Comment:
Though I'm not sure on killing `extendProjectDescriptionWithStandardLibraryCrates`. It uses `project` which is not available inside `CargoProjectDescription`, so there still has to be something that will call `withStdLib()` from outside and provide it with a `Library`.